internal/services/purview/purview_account_resource.go

```diff
@@ -43,55 +42,135 @@ func resourcePurviewAccount() *pluginsdk.Resource {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"catalog_endpoint": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
+
+			"guardian_endpoint": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
+
+			"scan_endpoint": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
+
+			"atlas_kafka_endpoint_primary_connection_string": {
+				Type:      pluginsdk.TypeString,
+				Computed:  true,
+				Sensitive: true,
+			},
+
+			"atlas_kafka_endpoint_secondary_connection_string": {
+				Type:      pluginsdk.TypeString,
+				Computed:  true,
+				Sensitive: true,
+			},
+
+			"tags": commonschema.Tags(),
+		},
 	}
 }
 
-func resourcePurviewAccountCreateUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+func resourcePurviewAccountCreate(d *pluginsdk.ResourceData, meta interface{}) error {
 	client := meta.(*clients.Client).Purview.AccountsClient
 	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
-	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
 	id := account.NewAccountID(subscriptionId, d.Get("resource_group_name").(string), d.Get("name").(string))
-	if d.IsNewResource() {
-		existing, err := client.Get(ctx, id)
-		if err != nil {
-			if !response.WasNotFound(existing.HttpResponse) {
-				return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
-			}
-		}
 
+	existing, err := client.Get(ctx, id)
+	if err != nil {
 		if !response.WasNotFound(existing.HttpResponse) {
-			return tf.ImportAsExistsError("azurerm_purview_account", id.ID())
+			return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
 		}
 	}
 
+	if !response.WasNotFound(existing.HttpResponse) {
+		return tf.ImportAsExistsError("azurerm_purview_account", id.ID())
+	}
+
 	purviewAccount := account.Account{
 		Properties: &account.AccountProperties{},
-		Location:   utils.String(azure.NormalizeLocation(d.Get("location").(string))),
-		Tags:       tags.Expand(d.Get("tags").(map[string]interface{})),
+		Location:   pointer.To(location.Normalize(d.Get("location").(string))),
+	}
+
+	if v, ok := d.GetOk("tags"); ok {
+		purviewAccount.Tags = tags.Expand(v.(map[string]interface{}))
 	}
 
 	expandedIdentity, err := identity.ExpandSystemOrUserAssignedMap(d.Get("identity").([]interface{}))
 	if err != nil {
 		return fmt.Errorf("expanding `identity`: %+v", err)
 	}
-
 	purviewAccount.Identity = expandedIdentity
 
-	publicNetworkAccessEnabled := account.PublicNetworkAccessDisabled
 	if d.Get("public_network_enabled").(bool) {
-		publicNetworkAccessEnabled = account.PublicNetworkAccessEnabled
+		purviewAccount.Properties.PublicNetworkAccess = pointer.To(account.PublicNetworkAccessEnabled)
+	} else {
+		purviewAccount.Properties.PublicNetworkAccess = pointer.To(account.PublicNetworkAccessDisabled)
```

Comment:
Similar question here on why this change was made. We tend to do it the was it was written so I'm wondering why this change was made

Reply:
I can put it back to how it was. My thought was trying to comply with only declaring variables when necessary, and making the code more concise / saving a line.